subprojects/core/src/integTest/groovy/org/gradle/api/internal/plugins/ApplyPluginBuildOperationIntegrationTest.groovy

```diff
@@ -67,63 +59,139 @@ class ApplyPluginBuildOperationIntegrationTest extends AbstractIntegrationSpec {
             null,
             null,
         ]
+    }
```

Comment:
This mechanism works different to ours. Here, we only convey the ID if the plugin was _applied_ via ID as opposed to _has_ an ID. My intention is to try and resolve this later.